src/api/sarifViewerApi.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from 'vscode';
+
+/**
+ * Options used when the SARIF viewer is opening logs.
+ */
+export interface OpenLogArguments {
+    /**
+     * Indicates whether to prompt the user when the
+     * SARIF log schema is needs to be upgraded to the latest one schema version.
+     * The default is true.
+     */
+    readonly promptUserForUpgrade?: boolean;
+
+    /**
+     * Indicates whether the SARIF viewer should open when there are no results in the logs to display.
+     * The default is true.
+     */
+    readonly openViewerOnNoResults?: boolean;
+}
+
+/**
+ * API exposed by the SARIF viewer extension.
+ */
+export interface Api {
```

Comment:
>Api [](start = 17, length = 3)

should there be some sort of version info embedded in this? How would backwards compatibility be supported in the future?

Reply:
There really doesn't need to be.

As long as you "add" new methods or properties you are good to go as far as JavaScript is concerned.

Heck, you can even add new parameters to the end of an API function and you'd still be good. :)

Point is, once someone is consuming your API, you can't ever really change the signatures and they must remain backward compatible.

---
In reply to: [409156312](https://github.com/microsoft/sarif-vscode-extension/pull/248#discussion_r409156312) [](ancestors = 409156312)